File: src/main/java/com/gorogoro/auth/authorization/domain/model/RefreshToken.java
package com.gorogoro.auth.authorization.domain.model;

import lombok.AccessLevel;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;

@Getter
@Builder
@AllArgsConstructor(access = AccessLevel.PRIVATE)
public class RefreshToken {
    private String refreshToken;
    private Long userId;
    private Long expiresAt;

    public static RefreshToken createRefreshToken(String refreshToken, Long userId, long durationMillis) {
        return RefreshToken.builder()
                .refreshToken(refreshToken)
                .userId(userId)
                .expiresAt(System.currentTimeMillis() + durationMillis)
                .build();
    }

    public boolean isExpired(long currentTime) {
        return currentTime > expiresAt;
    }
}

File: src/main/kotlin/com/gorogoro/auth/authorization/model/RefreshToken.kt
package com.gorogoro.auth.authorization.model

import java.time.Instant

class RefreshToken(
    val id: Long? = null,
    val userId: Long,
    refreshToken: String,
    refreshTokenExpire: Instant,
    ) {
    var refreshToken: String = refreshToken
        protected set

    var refreshTokenExpire: Instant = refreshTokenExpire
        protected set

    fun changeToken(token : String,refreshTokenExpire: Instant) {
        this.refreshToken = token
        this.refreshTokenExpire = refreshTokenExpire
    }

    fun isExpired(): Boolean {
        return Instant.now().isAfter(refreshTokenExpire)
    }
}
